packages/legend-query/src/components/QueryBuilderValueSpecificationEditor.tsx

```diff
@@ -617,11 +1172,22 @@ export const DateInstanceValueEditor = observer(
  */
 export const QueryBuilderValueSpecificationEditor: React.FC<{
   valueSpecification: ValueSpecification;
+  queryBuilderState: QueryBuilderState;
   graph: PureModel;
   expectedType: Type;
   className?: string | undefined;
+  setValueSpecification: (val: ValueSpecification | undefined) => void;
```

Comment:
Would be nice to have an optional parameter `resetValueSpecification` to enable the reset behavior though. If people don't specify it, we can make use of the default reset behavior (i.e. use the mock value for primitive type utility), else if they set it to `null`, we disable `reset`. Something like that. If you don't do it, leave this comment here so we can scrape it next time.